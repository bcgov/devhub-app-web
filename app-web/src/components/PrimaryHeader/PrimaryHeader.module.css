.PrimaryHeader {
    background-color: #036;
    border-bottom: 2px solid #fcba19;
    padding: 0 25px;
    color: #fff;
    display: flex;
    height: 65px;
    box-shadow: 0 1px 2px 2px rgba(0, 0, 0, .25);
    position: relative;
}


@media screen and (min-width: 600px) {
    .PrimaryHeader {
        padding: 0 65px 0 65px;
    }
   
}
  