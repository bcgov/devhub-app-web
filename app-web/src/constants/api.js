import config from '../config';

// config.json is used for Caddy templating during deployment
// to run locally, use .env.production
export const SSO_REALM_NAME = process.env.GATSBY_SSO_REALM_NAME || config.ssoRealmName;
export const SSO_CLIENT_ID = process.env.GATSBY_SSO_CLIENT_ID || config.ssoClientId;
export const SSO_BASE_URL = process.env.GATSBY_SSO_BASE_URL || config.ssoBaseUrl;
export const SEARCHGATE_API_URL = process.env.GATSBY_SEARCHGATE_API_URL || config.searchgateApiUrl;

// because this env variable GATBSY_ALGOLIA.. is made available at build, it will never utilize
// the caddy template string. This self invocated function will only allow the ennv variable to be used
// if the caddy template stringn has not been transformed
export const ALGOLIA_INDEX_SUFFIX = (() => {
  if (config.algoliaIndexSuffix.indexOf('{{.Env.ALGOLIA') === 0) {
    return process.env.GATSBY_ALGOLIA_INDEX_NAME_SUFFIX;
  }
  return config.algoliaIndexSuffix;
})();

export const GITHUB_URL = 'https://www.github.com';
