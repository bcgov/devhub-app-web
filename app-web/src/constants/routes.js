export const HOME_ROUTE = '/';
export const GITHUB_ISSUES_ROUTE = 'https://github.com/bcgov/devhub-web/issues';
