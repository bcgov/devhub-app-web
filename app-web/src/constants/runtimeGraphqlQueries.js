


import gql from 'graphql-tag';

export const ROCKET_GATE_QUERY  = gql`
  query DevHubQuery($queryString: String!) {
    search(searchString: $queryString) {
      message
      author
      time
      roomId
      id
    }
  }
`;