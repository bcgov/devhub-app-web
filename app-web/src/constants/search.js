// key of the query param in the url when searching ie ?q=
export const SEARCH_QUERY_PARAM = 'q';

export const SEARCH_FIELD_NAMES = [
  'title',
  'titles',
  'content',
  'contents',
  'description',
  'descriptions',
  'topicName',
  'topicname',
  'topicNames',
  'topicnames',
  'topic Name',
  'topic name',
  'topic Names',
  'topic names',
  'labels',
  'label',
  'author',
  'authors',
  'persona',
  'personas',
];

export const SEARCH_FIELD_MAPPING = {
  persona: { text: 'personas' },
  personas: { text: 'personas' },
  author: { text: 'author' },
  authors: { text: 'author' },
  label: { text: 'labels' },
  labels: { text: 'labels' },
  title: { text: 'title' },
  titles: { text: 'title' },
  content: { text: 'content' },
  contents: { text: 'content' },
  description: { text: 'description' },
  descriptions: { text: 'description' },
  topicName: { text: 'topicName' },
  topicname: { text: 'topicName' },
  topicNames: { text: 'topicName' },
  topicnames: { text: 'topicName' },
  'topic Name': { text: 'topicName' },
  'topic name': { text: 'topicName' },
  'topic Names': { text: 'topicName' },
  'topic names': { text: 'topicName' },
};

// federated search configurations
export const SEARCH_SOURCES = {
  rocketchat: 'rocketchat',
}

// this is a simple implementation of the search source 'toggle' feature
// this object acts like a reducer
export const SEARCH_SOURCE_INITIAL_STATE = {
  [SEARCH_SOURCES.rocketchat]: true,
};

export const SEARCH_SOURCE_CONFIG = {
  [SEARCH_SOURCES.rocketchat]: {
    maxResults: 10
  },
}