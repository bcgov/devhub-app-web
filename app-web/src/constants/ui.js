export const RESOURCE_TYPES = {
  COMPONENTS: 'Components',
  DOCUMENTATION: 'Documentation',
  SELF_SERVICE_TOOLS: 'Self-Service Tools',
  REPOSITORIES: 'Repositories',
  PEOPLE: 'People',
  EVENTS: 'Events',
};

export const SEARCH_RESOURCE_TYPES = {
  GITHUB_ISSUE: 'Github Issue',
  DOCUMIZE: 'Documize',
};

export const JOURNEY = 'Journey';
// not exactly a resource type but still an important identifier for a 'type' of devhub node
// this will change in devhub v3 when all nodes, resources/topics/and more are treated as
// equal
export const TOPICS = 'Topics';

export const RESOURCE_TYPES_LIST = Object.keys(RESOURCE_TYPES)
  .map(key => RESOURCE_TYPES[key])
  .concat(TOPICS)
  .concat(JOURNEY)
  .concat(SEARCH_RESOURCE_TYPES.GITHUB_ISSUE)
  .concat(SEARCH_RESOURCE_TYPES.DOCUMIZE);

export const PERSONAS_LIST = ['Developer', 'Designer', 'Product Owner'];

export const buttonTypes = ['primary', 'secondary', 'link'];

export const MAIN_NAVIGATION_BTN = 'dh-main-nav';
export const SEARCH_INPUT_ID = 'dh-search-input';
// for the card toggle component card limit prop

// react-scroll requires identifiers to scroll to individual elements
export const REACT_SCROLL = {
  CONFIG: {
    duration: 750,
    delay: 100,
    smooth: true,
  },
  ELEMENTS: {
    CARDS_CONTAINER: 'CARDS_CONTAINER',
  },
};

export const SEARCH = {
  INPUT: {
    maxLength: 40,
    id: SEARCH_INPUT_ID,
    name: `${SEARCH_INPUT_ID}.search`,
  },
};

// supported namespace for dynamically generating avatar images or icons
// this relates to the Avatar UI Component
export const AVATAR_NAMESPACES = {
  fontawesome: 'fontawesome',
  github: 'github',
};

export const EVENTS = {
  MAX_PAST_EVENTS: 9,
};

export const EVENT_TYPES = {
  eventbrite: 'eventbrite',
  meetup: 'meetup',
};

export const CARD_CAROUSEL = {
  mobile: {
    slidesPerPage: 1,
    slidesPerScroll: 1,
  },
  desktop: {
    slidesPerPage: 3,
    slidesPerScroll: 3,
  },
};

export const POPULAR_TOPIC_CONFIGURATION = {
  name: 'Popular',
  description: 'These resources have been getting some attention lately',
  maxNodes: 10, // this is calculated by measuring the 10 closest nodes relative the average page hit count
  minPageViews: 4,
};

export const DYNAMIC_TOPIC_PATHS = {
  popular: 'popular',
  featured: 'featured',
};

// featured content is identified by node.fields.title
// an expressed in order by this object
// this is a lookup table to easily identify which resources are considered featured
export const FEATURED_CONTENT = [
  'BC Developers Exchange',
  'Chat Channel Conventions',
  'Steps to join Pathfinder Rocket.Chat',
  '12 Factor Apps',
  'Progressive Web Apps',
  'Documize',
  'What is Pathfinder?',
];

export const FEATURE_TOPIC_CONFIGURATION = {
  name: 'Featured Resources',
  description: 'A selection of resources we think you will want to know about',
};

export const JOURNEY_TOPIC_VIEW_MODES = { card: 'cards', list: 'list' };
