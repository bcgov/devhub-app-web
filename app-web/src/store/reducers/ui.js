/*
Copyright 2018 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/
import * as actionTypes from '../actions/actionTypes';

const initialState = {
  selectedFilterOption: null,
  mainNavigationToggled: false,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case actionTypes.SET_SELECTED_FILTER_OPTION:
      return { ...state, selectedFilterOption: action.payload };
    case actionTypes.TOGGLE_MAIN_NAVIGATION:
      return { ...state, mainNavigationToggled: !state.mainNavigationToggled };
    default:
      return state;
  }
};

export default reducer;
