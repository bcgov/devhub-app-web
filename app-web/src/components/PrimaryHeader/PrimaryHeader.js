import React from 'react';
import PropTypes from 'prop-types';
import Banner from '../UI/Banner/Banner';
import Hamburger from '../UI/Hamburger/Hamburger';
// login not being implemented at this time
import Login from '../Auth/Login';
import styled from '@emotion/styled';
import { CUSTOM_BREAKPOINTS } from '../../constants/designTokens';

const Header = styled.header`
  background-color: ${({ theme }) => theme.colors.blue};
  border-bottom: 2px solid ${({ theme }) => theme.colors.yellow};
  padding: 0 12px 0 25px;
  color: #fff;
  display: flex;
  box-shadow: 0 1px 2px 2px rgba(0, 0, 0, 0.25);
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 50;
  height: 65px;
  @media screen and (min-width: 600px) {
    padding: 0 65px 0 65px;
  }
`;

const StyledHamburger = styled(Hamburger)`
  ${CUSTOM_BREAKPOINTS.navbar} {
    display: none;
  }
`;

const LoginWrapper = styled.div`
  display: none;

  ${CUSTOM_BREAKPOINTS.navbar} {
    display: inline-block;
  }
`;

const Panel = styled.div`
  flex-grow: 1;
  align-items: center;
  justify-content: flex-end;
  display: flex;
`;

export const PrimaryHeader = ({ showHamburger, hamburgerClicked }) => {
  return (
    <Header>
      <Banner />
      <Panel>
        <LoginWrapper>
          <Login />
        </LoginWrapper>
        {showHamburger ? <StyledHamburger clicked={hamburgerClicked} /> : null}
      </Panel>
    </Header>
  );
};

PrimaryHeader.propTypes = {
  showHamburger: PropTypes.bool,
  hamburgerClicked: PropTypes.func,
};

PrimaryHeader.defaultProps = {
  showHamburger: false,
  hamburgerClicked: () => undefined,
};

export default PrimaryHeader;
