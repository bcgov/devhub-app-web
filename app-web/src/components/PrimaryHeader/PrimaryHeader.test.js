import React from 'react';
import { shallow } from 'enzyme';
import { PrimaryHeader } from './PrimaryHeader';
import { LOGOUT_BTN_ID, LOGIN_BTN_ID } from '../../constants/ui';

describe('Primary Header Component', () => {
  test('it matches snapshot', () => {
    const wrapper = shallow(<PrimaryHeader />);
    expect(wrapper).toMatchSnapshot();
  });

  it('shows logout button when authenticated', () => {
    const wrapper = shallow(<PrimaryHeader isAuthenticated />);
    expect(wrapper.find(`#${LOGOUT_BTN_ID}`).length).toBe(1);
    expect(wrapper.find(`#${LOGIN_BTN_ID}`).length).toBe(0);
  });

  it('shows login button when not authenticated', () => {
    const wrapper = shallow(<PrimaryHeader isAuthenticated={false} />);
    expect(wrapper.find(`#${LOGOUT_BTN_ID}`).length).toBe(0);
    expect(wrapper.find(`#${LOGIN_BTN_ID}`).length).toBe(1);
  });
});
