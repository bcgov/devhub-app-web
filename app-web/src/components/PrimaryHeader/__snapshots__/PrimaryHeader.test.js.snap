// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary Header Component it matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PrimaryHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <GatsbyLink
        className="Logo"
        to="/"
      >
        <GovLogo />
        <h1>
          Developers Hub
        </h1>
      </GatsbyLink>,
      "className": "PrimaryHeader",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <GovLogo />,
          <h1>
            Developers Hub
          </h1>,
        ],
        "className": "Logo",
        "to": "/",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Developers Hub",
          },
          "ref": null,
          "rendered": "Developers Hub",
          "type": "h1",
        },
      ],
      "type": [Function],
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <GatsbyLink
          className="Logo"
          to="/"
        >
          <GovLogo />
          <h1>
            Developers Hub
          </h1>
        </GatsbyLink>,
        "className": "PrimaryHeader",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <GovLogo />,
            <h1>
              Developers Hub
            </h1>,
          ],
          "className": "Logo",
          "to": "/",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Developers Hub",
            },
            "ref": null,
            "rendered": "Developers Hub",
            "type": "h1",
          },
        ],
        "type": [Function],
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
